### ISSUE TITLE
Unable to download plugins using WP_PROXY_HOST and php7

### ISSUE THREAD

**Issue by dtolj** [wp-cli/wp-cli#3725]
When I set WP_PROXY_HOST in my wp-config.php and try "wp plugin install" it fails. As soon as I comment it out, the plugin is downloaded using the https_proxy env variable.
I need WP_PROXY_HOST for web interface as well. Whats causing wp to get confused?

I am using php-7.0.14 and curl-7.29 on rhel7.
I have tested it on php5.4 and it works but fails with php7.

Previously #3361

**Comment by danielbachhuber** [comment]
@dtolj Can you share:

* The specific error you see.
* Steps to reproduce the issue.
* Configuration details, including how you've set `WP_PROXY_HOST` and your PHP configuration?

**Comment by dtolj** [comment]
`wp plugin install bbpress --allow-root --debug`
```
Debug (bootstrap): No readable global config found (0.009s)
Debug (bootstrap): No project config found (0.009s)
Debug (bootstrap): No package autoload found to load. (0.048s)
Debug (bootstrap): ABSPATH defined: /var/www/html/practicalapps/ (0.048s)
Debug (bootstrap): Begin WordPress load (0.048s)
Debug (bootstrap): wp-config.php path: /var/www/html/practicalapps/wp-config.php (0.049s)
Debug (bootstrap): Loaded WordPress (0.461s)
Debug (bootstrap): Running command: plugin install (0.461s)
PHP Notice:  An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="https://wordpress.org/support/">support forums</a>. (WordPress could not establish a secure connection to WordPress.org. Please contact your server administrator.) in /var/www/html/practicalapps/wp-admin/includes/plugin-install.php on line 164
Notice: An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="https://wordpress.org/support/">support forums</a>. (WordPress could not establish a secure connection to WordPress.org. Please contact your server administrator.) in /var/www/html/practicalapps/wp-admin/includes/plugin-install.php on line 164
Warning: bbpress: An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="https://wordpress.org/support/">support forums</a>.
```

Steps to reproduce:
1) set Define('WP_PROXY_HOST','prxy123.com'); and Define('WP_PROXY_PORT', '8080');
2) run wp plugin install fails

Configs: wp-config.php
Define('WP_PROXY_HOST','prxy123.com');
Define('WP_PROXY_PORT', '8080');

env|grep http
http_proxy=http://prxy123.com:8080
https_proxy=http://prxy123.com:8080


No changes to php.ini

**Comment by danielbachhuber** [comment]
@dtolj Can you share your installed PHP extensions, and verify they're the same for both php-fpm and CLI?

**Comment by dtolj** [comment]
php70u-common-7.0.14-2.ius.el7.x86_64
php70u-fpm-7.0.14-2.ius.el7.x86_64
php70u-pdo-7.0.14-2.ius.el7.x86_64
php70u-fpm-nginx-7.0.14-2.ius.el7.noarch
php70u-json-7.0.14-2.ius.el7.x86_64
php70u-mysqlnd-7.0.14-2.ius.el7.x86_64
php70u-cli-7.0.14-2.ius.el7.x86_64

**Comment by danielbachhuber** [comment]
@dtolj Can you try installing `php-curl` and see if the problem is fixed?

> The behavior is different depending on whether php5-curl is installed. It ignores the proxy settings if I have it installed. Without it, I receive the error message expected in your new test case, even with proper proxy settings. The internal error message is:

From https://github.com/wp-cli/wp-cli/issues/3361#issuecomment-244621337

**Comment by dtolj** [comment]
php70u-curl was already installed, I solved the problem by updating wp cli to 1.0 and then it worked.
